### ISSUE TITLE
Add proxy support

### ISSUE THREAD

**Issue by sr258** [Lumieducation/Lumi#1309]

**Comment by Psych05** [comment]
This is essential. I work at a public authority responsible for IT at ~180 schools and we'd like to offer this software to our teachers. However, internet access is only available via proxy. This is extremely common in public computer networks.

**Comment by sr258** [comment]
Yes, we know about this need for corporate networks. I've added this feature to the underlying core library already, we now need to add it to the actual app.

After a year of constant staring at screens due to Covid lockdown, I took a bit of a break in the last few weeks, though.

How would you expect the proxy to be configured in your use case? I guess you want minimal configuration work, so auto detection would be best?

**Comment by Psych05** [comment]
Thanks for the reply. I'll keep an eye open for future updates then!

Regarding your question: In my opinion, the application should automatically use the settings in Windows Settings -> Network & Internet -> Proxy.

This is how well over 90% of Windows applications behave and it works just fine. The possibility to manually set a proxy specifically for a single application is usually only necessary if the application has very specific network-related requirements. I don't see that being the case here.

**Comment by swirly** [comment]
If proxy should be taken directly from system environment it would be great. All system store proxy parameters somewhere. It would lessen administrative work.

**Comment by sr258** [comment]
Yes, that's something that's certainly needed to ease configuration. The problem is that this requires an Electron browser window to be open as this is the only cross-OS way of getting the proxy settings in Electron. We need to add some kind of splash screen for that and I didn't have time to implement it. We cannot change the browser used by Lumi internally once the main window has been created.
